core/src/main/kotlin/com/exactpro/th2/read/db/core/impl/DataBaseServiceImpl.kt

```diff
@@ -46,36 +47,32 @@ class DataBaseServiceImpl(
 ) : DataBaseService {
     override fun executeQuery(
         dataSourceId: DataSourceId,
+        before: List<QueryId>,
         queryId: QueryId,
-        parameters: QueryParametersValues,
+        after: List<QueryId>,
+        parameters: QueryParametersValues
     ): Flow<TableRow> {
         LOGGER.info { "Executing query $queryId for $dataSourceId connection with $parameters parameters without default" }
         val dataSource: DataSource = dataSourceProvider[dataSourceId].dataSource
         val queryHolder: QueryHolder = queriesProvider[queryId]
+        val beforeQueryHolders: List<QueryHolder> = before.map(queriesProvider::get)
+        val afterQueryHolders: List<QueryHolder> = after.map(queriesProvider::get)
         val connection = dataSource.connection
         val finalParameters = queryHolder.defaultParameters.toMutableMap().apply {
             putAll(parameters)
         }
         LOGGER.trace { "Final execution parameters: $finalParameters" }
         val resultSet: ResultSet = try {
-            connection.prepareStatement(queryHolder.query).run {
-                queryHolder.parameters.forEach { (name, typeInfos) ->
-                    typeInfos.forEach { typeInfo ->
-                        val values = finalParameters[name]
-                        when {
-                            values == null -> set(typeInfo.index, null, typeInfo.type)
-                            values.size == 1 -> set(typeInfo.index, values.first(), typeInfo.type)
-                            else -> setCollection(typeInfo.index, connection.createArrayOf(typeInfo.type.name, values.toTypedArray()))
-                        }
-                    }
-                }
-                LOGGER.debug { "Native query: $this" }
-                executeQuery()
+            beforeQueryHolders.forEach { holder ->
+                execute(connection, holder, finalParameters)
```

Comment:
Can we wrap each `execute` call into a try-catch block and throw an exception describing which "before" query has failed? It should help to understand why the init query has failed.